src/analysis/vector/geometry_checker/qgsfeaturepool.h

```diff
@@ -24,39 +24,126 @@
 #include "qgis_analysis.h"
 #include "qgsfeature.h"
 #include "qgsspatialindex.h"
+#include "qgsvectorlayer.h"
 
 class QgsVectorLayer;
 
+/**
+ * \ingroup analysis
+ * A feature pool is based on a vector layer and caches features.
+ */
 class ANALYSIS_EXPORT QgsFeaturePool
```

Comment:
QgsFeaturePool could be a QgsFeatureSink/Source too, right?

Reply:
QgsFeatureSink should be alright.
QgsFeatureSource would be a bit more complicated because it only supports access by location (bbox request on spatial index) and request by feature id, so I wonder how to properly implement `getFeatures()`

Reply:
True. I guess you could proxy to the underlying layer/dataprovider, but maybe hold off till there's a requirement for this.

I'd still do feature sink, just because it is ;)

Reply:
I'll hold off for now. It's not very clear to decide what to do if there's a source with "only selected" and then a request comes in.